test/integration/mesos_test.go

```
@@ -0,0 +1,843 @@
+	}
+}
+
+func (cfg1 *testConfigData) runContainer(its *integTestSuite, c *C) {
+
+	if len(cfg1.containerName) == 0 {
+		return
+	}
+
+	// check if container exists
+	if oldCinfo, ok := containerInfo[cfg1.containerName]; ok {
+		cfg1.reqAttr = oldCinfo
+		return
+	}
+
+	containerName := cfg1.containerName
+	intLog.Infof("creating container: %s", containerName)
+
+	docker, _ := dockerclient.NewDockerClient("http://localhost:2385", nil)
+
+	// Create a container
+	docker.PullImage("alpine", nil)
+	containerConfig := &dockerclient.ContainerConfig{
+		Image: "alpine",
+		// self clean up after a few sec.
+		Cmd:             []string{"sleep", "60"},
+		AttachStdin:     false,
+		NetworkDisabled: true,
+		Env: []string{fmt.Sprintf("%s=%s", cniapi.EnvVarMesosAgent,
+			cfg1.agentIPAddr)},
+		Tty: false}
+
+	containerID, err := docker.CreateContainer(containerConfig,
+		containerName, nil)
+	assertNoErr(err, c, fmt.Sprintf("create container %s", containerName))
+
+	// Start the container
+	hostConfig := &dockerclient.HostConfig{}
+	err = docker.StartContainer(containerID, hostConfig)
+	assertNoErr(err, c, fmt.Sprintf("start container %s", containerName))
+
+	cfg1.reqAttr.CniContainerid = containerID
+	inspect, err := docker.InspectContainer(containerID)
+	assertNoErr(err, c, fmt.Sprintf("inspect container %s", containerName))
+	cfg1.reqAttr.CniNetns = fmt.Sprintf("/proc/%d/ns/net", inspect.State.Pid)
+	containerInfo[containerName] = cfg1.reqAttr
+	intLog.Infof("test container %s created %+v", containerID, containerInfo)
+	intLog.Infof("containers : %+v", containerInfo)
+}
+
+func (cfg1 *testConfigData) destroyContainer(its *integTestSuite, c *C) {
+	if len(cfg1.containerName) == 0 {
+		return
+	}
+
+	// check if container exists
+	if _, ok := containerInfo[cfg1.containerName]; !ok {
+		intLog.Warnf("no conainer with name %s", cfg1.containerName)
+		return
+	}
+
+	cinfo := &cfg1.reqAttr
+	docker, _ := dockerclient.NewDockerClient("http://localhost:2385", nil)
```

should we ignore the error?